01/implement-most-frequent-k-elements.checkpoint.md

```diff
@@ -0,0 +1,206 @@
+# Most Frequent K Elements
+
+<!-- prettier-ignore-start -->
+### !challenge
+* type: code-snippet
+* language: python3.6
+* id: c371a9d2-49b1-417b-abc4-3765e5cade18
+* title: Most Frequent K Elements
+* points: 3
+### !question
+
+Create a function, `most_frequent_k_elements`, that takes a non-empty array of integers and return the k most frequent elements. (Note: You may assume k is always valid, 1 ≤ k ≤ number of unique elements.)
+
+**Example 1:**
+
+```
+Input: nums = [1,1,1,2,2,3], k = 2
+Output: [1,2]
+```
+
+**Example 2:**
+```
+Input: nums = [1], k = 1
+Output: [1]
+```
+
+### !end-question
+### !placeholder
+
+```python
+def most_frequent_k_elements(arr,k):
+    """
+    A function that returns the k most frequent elements in a given array.
+  
+    Parameters:
+    arr (array): a non-empty array of integers
+    k (int): number of most frequent integers to return 
+  
+    Returns:
+    array[int]: 1D array that that returns the k most frequent elements
+    """
+    pass
+```
+### !end-placeholder
+### !tests
+```python
+import unittest
+from main import *
+
+class TestChallenge(unittest.TestCase):
+    def test_most_frequent_k_elements_nominal(self):
+        # Arrange
+        arr = [1,2,3,1,1,2,3]
+        k = 2
+
+        # Act
+        result = most_frequent_k_elements(arr, k)
+
+        # Assert
+        self.assertEqual(set(result), set([1,2]))
+
+    def test_most_frequent_k_elements_single_arr(self):
+        # Arrange
+        arr = [1]
+        k = 1 
+
+
+        # Act
+        result = most_frequent_k_elements(arr, k)
+
+        # Act
+        self.assertEqual(set(result), set([1]))
+
+    def test_most_frequent_k_elements_negative_number_case(self):
+        # Arrange
+        arr = [-1,-1,-1,-2,-5,-5,3,3,3,3,4]
+        k = 3
+
+        # Act
+        result = most_frequent_k_elements(arr, k)
+
+        # Assert
+        self.assertEqual(set(result), set([3,-1,-5]))
+
+
+
+```
+### !end-tests
+### !explanation
+
+An example of a working implementation:
+
+```python
+def most_frequent_k_elements(arr, k):
```

Comment:
I took it off because I was thinking it would spoil the answer for them, but this is for the example solutions not necessarily theirs. So I will put it back here, I feel like it would make more sense to see it here.